### DISCUSSION TITLE
Closing out old issues in GitHub

### DISCUSSION THREAD

**Discussion by JarroVGIT** [fastapi/fastapi#5014] [Questions] [answered]
First off, I don't know if this is the right place or if it is even my place to have this discussion. If it is not, my apologies.

I use FastAPI a lot and I love it. I wanted to give something back and try to spend some of my time in answering issues, or debugging issues that others have raised here at Github. However, I see that there are currently more than a thousand issues still open. 

I can confidently say that most of these are either;
- Not FastAPI related (not even close), or
- General questions that have been answered, or
- Resolved in newer releases of FastAPI (or Starlette/pydantic)

However, this amount of open issues does give a wrong impression of the quality of this project. And frankly, it does invite very low-quality questions to be asked and never be closed.

So, my question is; what can we do about this? My preference would be to have some contributors/moderators permissions to close old issues so we can keep a clean log of "real" open issues that require a fix or a feature update. I am wondering how others might look at this. I am more than willing to take part in such a team of moderators, and I am sure many are so. 

Any thoughts, @tiangolo ?

**Comment by odiseo0** [comment]
I think you would like to read this [issue](https://github.com/tiangolo/fastapi/issues/4263)
I like your proposal btw

**Comment by MrRawbin** [comment]
I like the proposal as well, and I have noted that you have been very active at giving solid responses. I think that you, and people like you, being given the permissions that you suggest would greatly benefit this project.

The same goes for PRs. At the very least, someone else should be able to triage them and probably also reject low quality/low effort PRs. As it stands right now, some very high quality PRs (at least in my opinion) are in the same pile as some low quality ones that I am 100% positive will never be merged, and which no one rejects right now.

**Reply by tiangolo** [reply]
There's actually no permissions needed to review PRs, but it requires people putting the effort and time to do that properly. I would definitely love to have more help with that, I documented the process to properly review PRs here: https://fastapi.tiangolo.com/help-fastapi/#review-pull-requests

**Comment by pythonweb2** [comment]
I noticed that Pydantic went the route of just closing any issues marked as a question, and instead encouraged asking questions in discussions. I would be in favor of FastAPI doing that as well.

**Comment by tiangolo** [comment, marked as answer]
Yep! I have been annoyed about the number of open "issues", when they are actually just questions.

I have been wanting to migrate to GitHub Discussions for a long while, I even had access to the beta, but it was missing two things, first, the API, to be able to compute FastAPI Experts (like you @JarroVGIT :bow: ): https://fastapi.tiangolo.com/fastapi-people/#experts

And then it was missing templates, because without templates people can then ask very open ended questions that take a lot of back and forth to just understand the actual question, the use case, and create the minimal reproduction example. That's why I still had that in issues, just because of the templates. In fact, I created the original feature request for that :sweat_smile: https://github.com/community/community/discussions/2838

But now, they just (finally) released templates for Discussions. :tada: 

The only missing piece was updating all the code to compute FastAPI Experts from issues and questions in Discussions. I just did that. :rocket: 

Now the recommended place to ask questions is Discussions: https://fastapi.tiangolo.com/help-fastapi/#ask-questions

---

About labeling and handling other issues and PRs, I would like to be able to give permissions for that to others, but I can only give full ownership permissions. And I also want to review all the issues myself too, to be able to see if I missed anything.

**Reply by JarroVGIT** [reply]
Great to hear!
